### DISCUSSION TITLE
Ignore Tooltip animation when other tooltips open

### DISCUSSION THREAD

**Discussion by pbarrowbyu121** [ariakit/ariakit#2155] [Questions] [answered]
The `timeout` prop for the `Tooltip` component is ignored when there's already an open tooltip in the page. Is there a way to animate the tooltip like a fade-in but ignore the animation if there's already an open tooltip in the page? This way subsequent tooltips won't fade-in just like subsequent tooltips won't have a timeout delay? Or maybe there's a prop that indicates if other tooltips are in the page so I can  use a conditional class/animation.

**Comment by diegohaz** [comment]
You can probably check the DOM for existing tooltips and change the animation accordingly. But we can also make that easier. Could you open a feature request?

**Comment by pbarrowbyu121** [comment, marked as answer]
For now I've used this workaround to determine if there is already a visible tooltip. I use this boolean var to then apply the `className` that animates the `Tooltip` component from `ariakit/tooltip`:

```import { Button } from "ariakit/button";
import { Tooltip, TooltipAnchor, useTooltipState } from "ariakit/tooltip";
import "./style.css";

let visibleTooltip: Boolean = false;
    document.querySelectorAll('[role="tooltip"]').forEach((element) => {
        if (window.getComputedStyle(element).display !== 'none') {
            visibleTooltip = true;
        }
    });

export default function Example() {
  const tooltip = useTooltipState();
  return (
    <>
      <TooltipAnchor state={tooltip} as={Button} className="button secondary">
        Hover or focus on me
      </TooltipAnchor>
      <Tooltip state={tooltip} className={`tooltip ${visibleTooltip ? "" : "animation-classname"}`}>
        Tooltip
      </Tooltip>
    </>
  );
}
